### ISSUE TITLE
How do you access route parameters?

### ISSUE THREAD

**Issue by vincentmorneau** [zurb/foundation-apps#635]
Let's say I have this route:

```
name: inbox.message
url: inbox/:id
```

And I'm accessing the page like this:
```
<a ui-sref="inbox.message({ id: 5 })">5th messages</a>
```

How do I access the value of parameter :id on the `inbox.message` page? It's just standing there at the moment.

I'm an angular newbie, so please be comprehensive. Most of what I'm reading leads to defining a `$stateProvider`, but I have no idea where to do that in Foundation for Apps.

![image](https://cloud.githubusercontent.com/assets/5391291/8144146/4468be78-119c-11e5-9ab5-a8fd59c7d22b.png)

**Comment by soumak77** [comment]
The answer you're looking for is the [$stateParams service](https://github.com/angular-ui/ui-router/wiki/URL-Routing#stateparams-service).  The link should give you all the info you need regarding how to use the params.

In your example, the id would be accessible via ```$stateParams.id```.  I don't believe the front matter provides a way to access the state params in your scope without wiring it through a controller.  I've seen others speak of a DefaultController which can be extended to add custom behavior, however, if I ever need any specific behavior from my controller I write my own versus extending the DefaultController provided by F4A, so you'd have to get info on how to do that from someone else.

In short, here is the code you'll need:
**Front Matter:**
```
---
name: inbox.message
url: inbox/:id
controller: InboxController
---
```

**Controller:**
```js
angular.module('application').controller('InboxController', 
  ['$scope', '$stateParams', function($scope, $stateParams) {
    $scope.msgId = $stateParams.id;
  }]
);
```

**View:**
```html
You're viewing message with id: {{ msgId }}
```

**Comment by gakimball** [comment]
@soumak77, you're the best! You seem to have adequately answered this question, but @vincentmorneau, if you need any more clarification, let us know.
